hive/src/test/java/com/netease/arctic/hive/io/HiveDataTestHelpers.java

```diff
@@ -19,111 +19,184 @@
+      this.usingHiveCommitProtocol = usingHiveCommitProtocol;
+      return this;
+    }
+
+    public WriterHelper orderedWrite(boolean orderedWrite) {
+      this.orderedWrite = orderedWrite;
+      return this;
+    }
+
+    public List<DataFile> writeChange(List<Record> records, ChangeAction action) {
+      AdaptHiveGenericTaskWriterBuilder builder =
+          AdaptHiveGenericTaskWriterBuilder.builderFor(table)
+              .withChangeAction(action)
+              .withTransactionId(txId);
+      if (orderedWrite) {
+        builder.withOrdered();
+      }
+      try (TaskWriter<Record> writer = builder.buildWriter(ChangeLocationKind.INSTANT)) {
+        return MixedDataTestHelpers.writeRecords(writer, records);
+      } catch (IOException e) {
+        throw new UncheckedIOException(e);
+      }
+    }
+
+    public List<DataFile> writeBase(List<Record> records) {
+      return writeRecords(BaseLocationKind.INSTANT, records);
+    }
+
+    public List<DataFile> writeHive(List<Record> records) {
+      return writeRecords(HiveLocationKind.INSTANT, records);
+    }
+
+    private List<DataFile> writeRecords(LocationKind writeLocationKind, List<Record> records) {
+      try (TaskWriter<Record> writer = newBaseWriter(writeLocationKind)) {
+        return MixedDataTestHelpers.writeRecords(writer, records);
+      } catch (IOException e) {
+        throw new UncheckedIOException(e);
+      }
+    }
 
-  public static List<DataFile> writeBaseStore(
-      ArcticTable table,
-      long txId,
-      List<Record> records,
-      boolean orderedWrite,
-      boolean writeHiveLocation,
-      String hiveLocation) {
-    AdaptHiveGenericTaskWriterBuilder builder = AdaptHiveGenericTaskWriterBuilder.builderFor(table);
-    if (table.isKeyedTable()) {
-      builder.withTransactionId(txId);
+    private TaskWriter<Record> newBaseWriter(LocationKind writeLocationKind) {
+      AdaptHiveGenericTaskWriterBuilder builder =
+          AdaptHiveGenericTaskWriterBuilder.builderFor(table);
+      if (table.isKeyedTable()) {
+        builder.withTransactionId(txId);
+      }
+      if (orderedWrite) {
+        builder.withOrdered();
+      }
+      if (customHiveLocation != null) {
+        builder.withCustomHiveSubdirectory(customHiveLocation);
+      }
+      if (this.usingHiveCommitProtocol != null) {
+        builder.hiveConsistentWrite(this.usingHiveCommitProtocol);
+      }
+      return builder.buildWriter(writeLocationKind);
     }
-    if (orderedWrite) {
-      builder.withOrdered();
+  }
+
+  /** Simulate the consistent-write-commit process to convert a hidden file to a visible file. */
+  public static List<DataFile> applyConsistentWriteFiles(ArcticTable table, List<DataFile> files) {
+    if (!TablePropertyUtil.hiveConsistentWriteEnabled(table.properties())) {
+      return files;
     }
-    if (hiveLocation != null) {
-      builder.withCustomHiveSubdirectory(hiveLocation);
+    List<DataFile> nonHiveFiles = Lists.newArrayList();
```

Comment:
Maybe we can start a file scan to get files rather than changing the written files.